core/src/main/scala/kafka/log/Log.scala

```diff
@@ -302,7 +312,13 @@ class Log(@volatile var dir: File,
     // create the log directory if it doesn't exist
     Files.createDirectories(dir.toPath)
     val checkpointFile = new LeaderEpochCheckpointFile(LeaderEpochFile.newFile(dir), logDirFailureChannel)
-    new LeaderEpochFileCache(topicPartition, logEndOffset _, checkpointFile)
+    val cache = new LeaderEpochFileCache(topicPartition, logEndOffset _, checkpointFile)
+
+    if (!supportsLeaderEpoch && cache.epochEntries.nonEmpty) {
```

Comment:
I'm not sure we should access `epochEntries` directly since it bypasses the lock in `LeaderEpochFileCache`.

Reply:
You're right. I didn't want to use the `UNDEFINED_EPOCH` sentinel value but that seems like the better way to go